cpp/src/strings/convert.cu

```diff
@@ -0,0 +1,257 @@
+    char* d_chars;
+
+    __device__ void operator()(size_type idx)
+    {
+        if( d_column.is_null(idx) )
+            return;
+        IntegerType value = d_column.element<IntegerType>(idx);
+        char* d_buffer = d_chars + d_offsets[idx];
+        if( value==0 )
+        {
+            *d_buffer = '0';
+            return;
+        }
+        bool sign = value < 0;
+        if( sign )
+            value = -value;
+        constexpr IntegerType base = 10;
+        char* ptr = d_buffer;
+        while( value > 0 )
+        {
+            *ptr++ = '0' + (value % base);
+            value = value/base;
+        }
+        if( sign )
+            *ptr++ = '-';
+        size_type length = static_cast<size_type>(ptr-d_buffer);
+        // numbers are backwards, reverse the string
+        for( size_type j=0; j<(length/2); ++j )
+        {
+            char ch1 = d_buffer[j];
+            char ch2 = d_buffer[length-j-1];
+            d_buffer[j] = ch2;
+            d_buffer[length-j-1] = ch1;
+        }
+    }
+};
+
+/**
+ * @brief This dispatch method is for converting integers into strings.
+ * The template function declaration ensures only integer types are used.
+ */
+struct dispatch_from_integers_fn
+{
+    template <typename IntegerType, std::enable_if_t<std::is_integral<IntegerType>::value>* = nullptr>
+    std::unique_ptr<cudf::column> operator()( column_view& integers,
+                                              rmm::mr::device_memory_resource* mr,
+                                              cudaStream_t stream ) const noexcept
+    {
+        size_type strings_count = integers.size();
+        auto execpol = rmm::exec_policy(0);
+        auto column = column_device_view::create(integers, stream);
+        auto d_column = *column;
+
+        // copy the null mask
+        rmm::device_buffer null_mask;
+        cudf::size_type null_count = d_column.null_count();
+        if( d_column.nullable() )
+            null_mask = rmm::device_buffer( d_column.null_mask(),
+                                            gdf_valid_allocation_size(strings_count),
+                                            stream, mr);
+        // build offsets column
+        auto offsets_transformer_itr = thrust::make_transform_iterator( thrust::make_counting_iterator<int32_t>(0),
+            integer_to_string_size_fn<IntegerType>{d_column} );
+        auto offsets_column = detail::make_offsets_child_column(offsets_transformer_itr,
+                                                                offsets_transformer_itr+strings_count,
+                                                                mr, stream);
+        auto offsets_view = offsets_column->view();
+        auto d_new_offsets = offsets_view.template data<int32_t>();
+
+        // build chars column
+        size_type bytes = thrust::device_pointer_cast(d_new_offsets)[strings_count];
+        auto chars_column = detail::create_chars_child_column( strings_count, null_count, bytes, mr, stream );
```

Comment:
Why does `create_chars_child_column` accept a `null_count`? I thought the chars child can never contain nulls?

Reply:
This is to handle an edge case -- explained at the bottom of slide 8 of my presentation and in the comment inside the function around https://github.com/rapidsai/cudf/blob/4f60daea25897792030a77e043095d4239a37f4e/cpp/src/strings/utilities.cu#L129
If all non-null strings are empty strings we need a minimal chars column.